### PR TITLE
Fix duplicated scripts and prep to publish dwds

### PR DESCRIPTION
**Pull request by annagrin**
Make inspector initializer wait on the same future as
getScripts call in ChromeProxyService to make sure
we populate caches only once.

Prep to publish to support file picker work in devtools.

Closes: https://github.com/dart-lang/webdev/issues/1413

### CODE REVIEW

**Comment by nshahan** [comment]
I'm trying to see how this would be different but I don't see when the cache is ever populated. How does that happen?

File: dwds/lib/src/debugging/inspector.dart [hunk 1]
```diff
@@ -87,7 +87,7 @@ class AppInspector extends Domain {
     isolate.libraries.addAll(libraries);
     await DartUri.recordAbsoluteUris(libraries.map((lib) => lib.uri));
 
-    var scripts = await _getScripts();
+    var scripts = await scriptRefs;
```
Comment (grouma) [inline dwds/lib/src/debugging/inspector.dart:90]: We should probably add a comment to `_getScripts` indicating that callers should use `scriptRefs` instead. Or potentially rename `_getScripts` to `_fetchScripts` or something to make it more clear.

**Review by grouma — APPROVED** [review]

File: dwds/lib/src/debugging/inspector.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/inspector.dart:90]: sure, will do

Comment (annagrin) [reply dwds/lib/src/debugging/inspector.dart:92]: I rewrote it a bit so we cannot accidentally call this function twice.

File: dwds/lib/src/debugging/inspector.dart [hunk 2]
```diff
@@ -483,38 +484,36 @@ function($argsString) {
...
-  Future<List<ScriptRef>> _getScripts() async {
-    await _populateScriptCaches();
-    return _scriptRefsById.values.toList();
-  }
-
   /// Request and cache <ScriptRef>s for all the scripts in the application.
   ///
   /// This populates [_scriptRefsById], [_scriptIdToLibraryId] and
   /// [_serverPathToScriptRef]. It is a one-time operation, because if we do a
   /// reload the inspector will get re-created.
```
Comment (nshahan) [inline dwds/lib/src/debugging/inspector.dart:491]: The comment needs a description of the return value now.

**Review by nshahan — APPROVED** [review]
One comment but LGTM.

**Comment by annagrin** [comment]
> I'm trying to see how this would be different but I don't see when the cache is ever populated. How does that happen?

The cache is populated on first `scriptRefs` call during inspector initialization.

**Pull request merged**